subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -1060,28 +1136,64 @@ static int flash_callback(uint8_t *buf, size_t len, size_t offset)
 void *flash_callback;
 #endif
 
-static int open_storage(uint32_t offset, bool preserve)
+static int open_flash(void)
 {
 	int err;
-	const struct device *flash_dev = DEVICE_DT_GET(DT_NODELABEL(flash_controller));
-	uint32_t block_offset = 0;
 
-	LOG_DBG("storage name: %s", flash_dev->name);
+#if defined(CONFIG_NRF_CLOUD_PGPS_STORAGE_PARTITION)
+	flash_dev = FLASH_AREA_DEVICE(PGPS);
+	flash_area_id = FLASH_AREA_ID(PGPS);
+#elif defined(CONFIG_NRF_CLOUD_PGPS_STORAGE_MCUBOOT_SECONDARY)
+	flash_dev = FLASH_AREA_DEVICE(MCUBOOT_SECONDARY);
+	flash_area_id = FLASH_AREA_ID(MCUBOOT_SECONDARY);
+#else
+	flash_dev = DEVICE_DT_GET(DT_NODELABEL(flash_controller));
+	flash_area_id = FLASH_AREA_ID(DT_NODELABEL(flash_controller));
+#endif
+
+	err = flash_area_open(flash_area_id, &flash);
+	if (err) {
+		flash = NULL;
+		LOG_WRN("Cannot access predictions using flash_area; default to direct reads: %d",
```

Comment:
I think `flash_area_read` will straight up fail if `flash==NULL`, right? Which part of the code checks for this and defaults to direct reads, and what is meant by direct reads?

Or am I mistaken?

Reply:
Removing -- development code not needed anymore